pkg/crc/cluster/cert-renewal.go

```diff
@@ -2,53 +2,53 @@ package cluster
 
 import (
 	"fmt"
-	"regexp"
+	"strings"
 	"time"
 
 	"github.com/code-ready/crc/pkg/crc/errors"
 	"github.com/code-ready/crc/pkg/crc/logging"
 	"github.com/code-ready/crc/pkg/crc/oc"
 )
 
-func waitForPendingCsrs(ocConfig oc.Config) error {
+func waitForPendingCsrs(ocConfig oc.Config, signerName string) error {
 	waitForPendingCsr := func() error {
 		output, _, err := ocConfig.RunOcCommand("get", "csr")
 		if err != nil {
 			return &errors.RetriableError{Err: err}
 		}
-		matched, err := regexp.MatchString("Pending", output)
-		if err != nil {
-			return &errors.RetriableError{Err: err}
-		}
-		if !matched {
-			return &errors.RetriableError{Err: fmt.Errorf("No Pending CSR")}
+		if strings.Contains(output, "Pending") && strings.Contains(output, signerName) {
+			return nil
 		}
-		return nil
+		return &errors.RetriableError{Err: fmt.Errorf("No Pending CSR with signerName %s", signerName)}
 	}
 
 	return errors.RetryAfter(8*time.Minute, waitForPendingCsr, time.Second*5)
 }
 
-func RegenerateCertificates(ocConfig oc.Config) error {
+func WaitAndApprovePendingCSRs(ocConfig oc.Config, client, server bool) error {
 	/* 2 CSRs to approve, one right after kubelet restart, the other one a few dozen seconds after
 	approving the first one
 	- First one is requested by system:serviceaccount:openshift-machine-config-operator:node-bootstrapper
 	- Second one is requested by system:node:<node_name> */
-	if err := waitForPendingCsrs(ocConfig); err != nil {
-		logging.Debugf("Error waiting for first pending (node-bootstrapper) CSR: %v", err)
-		return err
-	}
-	if err := ApproveNodeCSR(ocConfig); err != nil {
-		logging.Debugf("Error approving first pending (node-bootstrapper) CSR: %v", err)
-		return err
-	}
-	if err := waitForPendingCsrs(ocConfig); err != nil {
-		logging.Debugf("Error waiting for second pending (system:node) CSR: %v", err)
-		return err
+	if client {
+		if err := waitForPendingCsrs(ocConfig, "kubernetes.io/kube-apiserver-client-kubelet"); err != nil {
+			logging.Debugf("Error waiting for first pending (node-bootstrapper) CSR: %v", err)
+			return err
+		}
+		if err := ApproveNodeCSR(ocConfig, "kubernetes.io/kube-apiserver-client-kubelet"); err != nil {
+			logging.Debugf("Error approving first pending (node-bootstrapper) CSR: %v", err)
```

Comment:
It's no longer first/second, but client/server CSRs. I guess it would be possible to only get the second CSR for example in some scenarios.

Reply:
yes changed!